src/mgmt/srv_pool.c

```diff
@@ -361,8 +362,16 @@ ds_mgmt_pool_target_update_state(uuid_t pool_uuid, d_rank_list_t *svc_ranks,
 		}
 	}
 
-	rc = dsc_pool_svc_update_target_state(pool_uuid, svc_ranks, mgmt_ps_call_deadline(),
-					      target_addrs, state, skip_rf_check);
+	deadline = mgmt_ps_call_deadline();
+
+again:
+	rc = dsc_pool_svc_update_target_state(pool_uuid, svc_ranks, deadline, target_addrs, state,
+					      skip_rf_check);
+	if (rc == -DER_AGAIN && state == PO_COMP_ST_UP && daos_getmtime_coarse() < deadline) {
+		D_WARN("Retry incremental reintegration for pool " DF_UUID " because of race\n",
+		       DP_UUID(pool_uuid));
```

Comment:
should we just return EAGAIN to client and client can retry? I'm a little nervous about the local retry.

Reply:
Before timeout, local retry is general process for our current POOL_REINT/EXTENT/EXCLUDE operation. Please check dsc_pool_svc_call().